sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java

```diff
@@ -666,35 +667,59 @@ public void rotateKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
             .verifyComplete();
     }
 
-    private void pollOnKeyDeletion(String keyName) {
-        int pendingPollCount = 0;
-        while (pendingPollCount < 30) {
-            DeletedKey deletedKey = null;
-            try {
-                deletedKey = client.getDeletedKeyWithResponse(keyName).block().getValue();
-            } catch (ResourceNotFoundException e) {
-            }
-            if (deletedKey == null) {
-                sleepInRecordMode(2000);
-                pendingPollCount += 1;
-            } else {
-                return;
-            }
-        }
-        System.err.printf("Deleted Key %s not found \n", keyName);
+    /**
+     * Tests that a {@link CryptographyAsyncClient} can be created for a given key and version using a
+     * {@link KeyAsyncClient}.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void getCryptographyAsyncClientWithKeyVersion(HttpClient httpClient, KeyServiceVersion serviceVersion) {
+        createKeyAsyncClient(httpClient, serviceVersion);
+
+        CryptographyAsyncClient cryptographyAsyncClient =
+            client.getCryptographyAsyncClient("myKey","6A385B124DEF4096AF1361A85B16C204");
+
+        assertNotNull(cryptographyAsyncClient);
```

Comment:
You should consider using it in at least one test. I, too, did something similar assuming it was so straight forward...how could I screw it up? Turns out, I did. Well, it was more about nuanced behavior of the code I was taking advantage of from our `KeyResolver` implementation. Turns out: it didn't work. I caught it second-hand and, once I did, added quite a few tests.